app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala

```diff
@@ -56,21 +55,23 @@ case class OrganisationDetails(
         details.registration.registrationStatus == "REGISTRATION_NOT_CALLED" && details.businessVerificationStatus == "FAIL"
     )
 
-  def businessPartnerIdPresent(): Boolean =
+  def businessPartnerId(): Option[String] =
     organisationType match {
       case Some(UK_COMPANY) =>
-        incorporationDetails.exists(_.registration.registeredBusinessPartnerId.isDefined)
+        incorporationDetails.fold[Option[String]](None)(_.registration.registeredBusinessPartnerId)
```

Comment:
I might be wrong but doesn't Option.map() do want you want here?

Reply:
Or maybe `flatMap()`?

Reply:
Agree with Ryan.   This should do the trick
```

  def businessPartnerId(): Option[String] =
    organisationType match {
      case Some(UK_COMPANY) =>
        incorporationDetails.flatMap(_.registration.registeredBusinessPartnerId)
      case Some(SOLE_TRADER) =>
        soleTraderDetails.flatMap(_.registration.registeredBusinessPartnerId)
      case GENERAL_PARTNERSHIP =>
        partnershipDetails.flatMap(
          _.generalPartnershipDetails.flatMap(_.registration.registeredBusinessPartnerId)
        )
      case SCOTTISH_PARTNERSHIP =>
        partnershipDetails.flatMap(
          _.scottishPartnershipDetails.flatMap(_.registration.registeredBusinessPartnerId)
        )
      case _ => None
    }
```

Reply:
Actually, that is not quite right.  Partnership needs splitting up as before

Reply:
Addressed with use of flatMap